internal/pkg/uploader/upload_test.go

```diff
@@ -0,0 +1,273 @@
+		action+"."+agent,    // document ID
+		mock.Anything,       // ES document
+		mock.Anything,       // bulker options
+	).Return("", nil)
+
+	c, err := cache.New(config.Cache{NumCounters: 100, MaxCost: 100000})
+	require.NoError(t, err)
+	u := New(nil, fakeBulk, c, int64(size), time.Hour)
+	_, err = u.Begin(context.Background(), data)
+	assert.NoError(t, err)
+
+	payload, ok := fakeBulk.Calls[0].Arguments[3].([]byte)
+	assert.Truef(t, ok, "argument to es create should be byte slice")
+
+	j := make(JSDict)
+	err = json.Unmarshal(payload, &j)
+	assert.NoError(t, err)
+
+}
+
+func TestUploadBeginCalculatesCorrectChunkCount(t *testing.T) {
+	fakeBulk := itesting.NewMockBulk()
+
+	fakeBulk.On("Create",
+		mock.Anything, // match context.Context
+		mock.Anything, // index
+		mock.Anything, // document ID
+		mock.Anything, // ES document
+		mock.Anything, // bulker options
+	).Return("", nil)
+
+	tests := []struct {
+		FileSize      int64
+		ExpectedCount int
+		Name          string
+	}{
+		{10, 1, "Tiny files take 1 chunk"},
+		{MaxChunkSize, 1, "Precisely 1 chunk size bytes should fit in 1 chunk"},
+		{MaxChunkSize + 1, 2, "ChunkSize+1 bytes takes 2 chunks"},
+		{MaxChunkSize * 3.5, 4, "3.5x chunk size fits in 4 chunks due to remainder"},
+		{7534559605, 1797, "7.5Gb file"},
+	}
+
+	c, err := cache.New(config.Cache{NumCounters: 100, MaxCost: 100000})
+	require.NoError(t, err)
+	u := New(nil, fakeBulk, c, MaxChunkSize*3000, time.Hour)
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			data := makeUploadRequestDict(map[string]interface{}{
+				"file.size": tc.FileSize,
+			})
+			info, err := u.Begin(context.Background(), data)
+			assert.NoError(t, err)
+			assert.Equal(t, tc.ExpectedCount, info.Count)
+		})
+	}
+}
+
+func TestUploadBeginMaxFileSize(t *testing.T) {
+	tests := []struct {
+		UploadSizeLimit int64
+		FileSize        int64
+		ShouldError     bool
+		Name            string
+	}{
+		{500, 800, true, "800 is too large"},
+		{800, 500, false, "file within limits"},
+		{1024, 1023, false, "1-less than limit"},
+		{1024, 1024, false, "file is exactly limit"},
+		{1024, 1025, true, "file is 1 over limit"},
+	}
+
+	fakeBulk := itesting.NewMockBulk()
+	fakeBulk.On("Create",
+		mock.Anything, // context.Context
+		mock.Anything, // index
+		mock.Anything, // document ID
+		mock.Anything, // ES document
+		mock.Anything, // bulker options
+	).Return("", nil)
+
+	c, err := cache.New(config.Cache{NumCounters: 100, MaxCost: 100000})
+	require.NoError(t, err)
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			u := New(nil, fakeBulk, c, tc.UploadSizeLimit, time.Hour)
+			data := makeUploadRequestDict(map[string]interface{}{
+				"file.size": tc.FileSize,
+			})
+			_, err := u.Begin(context.Background(), data)
+			if tc.ShouldError {
+				assert.ErrorIs(t, err, ErrFileSizeTooLarge)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+/*
+func TestUploadRejectsMissingRequiredFields(t *testing.T) {
```

Comment:
are you planning to fix these tests or remove the code?

Reply:
these are staying in, just need updating